### PR TITLE
Added setting the required size of the number of cells when loading data through LoadFunctions

### PR DESCRIPTION
**Pull request by LIFEfreedom**
I noticed that by default ExcelWorksheet creates a CellStoreValue with an empty List <ExcelValue> inside. When you add the first cell, the List size becomes 4, and then doubles every time the capacity is less than the required number of elements ([can be seen in the source code of List](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs)). Increasing the size of the List entails the creation of a new array, followed by copying the old array into a new one, and the more such changes, the more work the garbage collector will have.
When using data loading via LoadFunctions, we know exactly how many cells we need to load all the data, so we can preset the required size and use just one array

### CODE REVIEW

File: src/EPPlus/FormulaParsing/EpplusExcelDataProvider.cs [hunk 1]
```diff
@@ -386,6 +386,7 @@ public override IRangeInfo GetRange(string worksheet, int row, int column, strin
             //SetCurrentWorksheet(addr.WorkSheet); 
             var wsName = string.IsNullOrEmpty(addr.WorkSheetName) ? _currentWorksheet.Name : addr.WorkSheetName;
             var ws = _package.Workbook.Worksheets[wsName];
+
```
Comment (LIFEfreedom) [inline src/EPPlus/FormulaParsing/EpplusExcelDataProvider.cs:389]: result of resolve conflict

**Comment by LIFEfreedom** [comment]
@JanKallman Any feedback?

**Comment by JanKallman** [comment]
> @JanKallman Any feedback?

Sorry for the delayed answer. Sounds like a good ideas. I will look at the pull requests in more details prior to the release of 5.7, in a week or so, so I will get back to you.

**Comment by JanKallman** [comment]
Thanks

**Pull request merged**